File: modules/user/server-scala/src/main/scala/model/auth/AuthInput.scala
package model.auth

import model.AuthCertificateInput
import model.auth.facebook.AuthFacebookInput
import model.auth.github.AuthGitHubInput
import model.auth.google.AuthGoogleInput
import model.auth.linkedin.AuthLinkedInInput

case class AuthInput(
    certificate: Option[AuthCertificateInput],
    facebook: Option[AuthFacebookInput],
    google: Option[AuthGoogleInput],
    github: Option[AuthGitHubInput],
    linkedin: Option[AuthLinkedInInput]
)

File: modules/user/server-java/src/main/java/com/sysgears/user/dto/input/auth/AuthInput.java
package com.sysgears.user.dto.input.auth;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Optional;

@Data
@NoArgsConstructor
public class AuthInput {
    private final Optional<AuthCertificateInput> certificate = Optional.empty();
    private final Optional<AuthFacebookInput> facebook = Optional.empty();
    private final Optional<AuthGoogleInput> google = Optional.empty();
    private final Optional<AuthGitHubInput> github = Optional.empty();
    private final Optional<AuthLinkedInInput> linkedin = Optional.empty();
}
